src/auth/src/credentials.rs

```diff
@@ -137,6 +139,78 @@ impl Credentials {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct AccessTokenCredentials {
+    // We use an `Arc` to hold the inner implementation.
+    //
+    // AccessTokenCredentials may be shared across threads (`Send + Sync`), so an `Rc`
+    // will not do.
+    //
+    // They also need to derive `Clone`, as the
+    // `gax::http_client::ReqwestClient`s which hold them derive `Clone`. So a
+    // `Box` will not do.
+    inner: Arc<dyn dynamic::AccessTokenCredentialsProvider>,
+}
+
+impl<T> std::convert::From<T> for AccessTokenCredentials
+where
+    T: crate::credentials::AccessTokenCredentialsProvider + Send + Sync + 'static,
+{
+    fn from(value: T) -> Self {
+        Self {
+            inner: Arc::new(value),
+        }
+    }
+}
+
+impl AccessTokenCredentials {
+    pub async fn token(&self) -> Result<AccessToken> {
+        self.inner.token().await
+    }
+}
+
+impl CredentialsProvider for AccessTokenCredentials {
+    async fn headers(&self, extensions: Extensions) -> Result<CacheableResource<HeaderMap>> {
+        self.inner.headers(extensions).await
+    }
+
+    async fn universe_domain(&self) -> Option<String> {
+        self.inner.universe_domain().await
+    }
+}
+
+pub struct AccessToken {
```

Comment:
nit: documentation for public types, but I think you are just sketching the interfaces for now.